src/NuGet.Core/NuGet.Packaging/Signing/Timestamp/Rfc3161TimestampProvider.cs

```diff
@@ -145,6 +145,39 @@ public byte[] TimestampData(TimestampRequest request, ILogger logger, Cancellati
             }
         }
 
+        private static void ValidateTimestampCms(SigningSpecifications spec, SignedCms timestampCms)
+        {
+            var signerInfo = timestampCms.SignerInfos[0];
+            try
+            {
+                signerInfo.CheckSignature(verifySignatureOnly: true);
+            }
+            catch
```

Comment:
Can you catch the exception and pass it to `TimestampException`'s constructor as an inner exception?